src/app/(main)/dashboard/components/voyage-dashboard/getDashboardData.ts

```diff
@@ -0,0 +1,108 @@
+import { format, parseISO } from "date-fns";
+import { fetchSprints } from "@/app/(main)/my-voyage/[teamId]/sprints/components/RedirectToCurrentSprintWrapper";
+import { Sprint } from "@/store/features/sprint/sprintSlice";
+import { getCurrentSprint } from "@/utils/getCurrentSprint";
+import { getCurrentVoyageData } from "@/utils/getCurrentVoyageData";
+import { User } from "@/store/features/user/userSlice";
+import { fetchMeeting } from "@/app/(main)/my-voyage/[teamId]/sprints/components/SprintWrapper";
+import { getUser } from "@/utils/getUser";
+
+interface GetDashboardDataResponse {
+  currentSprintNumber: number | null;
+  sprintsData: Sprint[];
+  user: User | null;
+  meetingsData: EventList[];
+  errorMessage?: string;
+}
+
+export type EventList = {
+  title: string;
+  date: string;
+  link: string;
+};
+
+export const getDashboardData = async (): Promise<GetDashboardDataResponse> => {
+  let sprintsData: Sprint[] = [];
+
+  const [user, error] = await getUser();
+
+  const teamMember = user?.voyageTeamMembers.find(
+    (voyage) => voyage.voyageTeam.voyage.status.name === "Active",
+  );
+
+  const teamId = teamMember?.voyageTeamId;
+
+  if (!teamId) {
+    return {
+      currentSprintNumber: null,
+      sprintsData: [],
+      user: null,
+      meetingsData: [],
+    };
+  }
```

Comment:
This part shouldn't be necessary? it's all covered in the getcurrentvoyagedata function.

Reply:
Maybe I've done something wrong, but I think we need the teamId (so the teamMember) to call the getCurrentVoyageData method

Reply:
I just realized the dashboard isn't making use of path parameters. It might be easier to make use of that there too, but it gets tricky because we would need to cover for dashboard page for both voyage members (make us of path parameters) and non voyage members (no path parameters).

Reply:
Maybe we can create two pages, one with the teamId and one without, for example:

[teamId]/dashboard --> voyage members
/dashboard --> non voyage members

On the dashboard, we'll display different data based on the presence of teamId. All components will be shared between the two pages. What do you think?

Reply:
That sounds like a good approach.

Reply:
Now we have the two pages based on the teamId, but I don't know how we can redirect the user. We have this in the next.config file:

```
    async redirects() {
        return [
          {
            source: '/',
            destination: '/dashboard',
            permanent: true,
          },
        ];
      },
```

Maybe we can use the nextjs middleware to check if the user is a voyage user and then redirect to the /[teamId]/dashboard page?

Reply:
I don't think you can use middleware to check that because it'd involve having to make api calls. I tried to do something else that involved this and it didn't work. What you can do though is something similar to what Jane did. Just have a page.tsx for the voyage dashboard for non voyage members (and any components needed), and then a [teamId] folder where you'll have all the stuff for voyage dashboard. In the root dashboard folder you can do the check to see if they're voyage member or not and redirect if they are. That might be hard to understand, let me know if you didn't understand / need clarification.

Reply:
I ran a test in my last commit. Please let me know if this aligns with your expectations. Thank you for your support!